### ISSUE TITLE
AdGuard cuts off network connection to Custom DNS on iOS

### ISSUE THREAD

**Issue by Batman2814** [AdguardTeam/AdguardForiOS#2293]
### Please answer the following questions for yourself before submitting an issue

- [X] Filters were updated before reproducing an issue
- [X] I checked the [knowledge base](https://adguard.com/kb/) and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard version

4.5.2

### Environment

- OS: 17.3.1
- Device: iPhone 17 plus

### Advanced protection for Safari

- [X] yes, I do

### Which DNS server do you use?

Custom DNS

### DNS protocol

DNS-over-QUIC

### Custom DNS

AdGuard Private DNS

### DNS filtering

- [X] yes, I do

### Custom DNS filter

Na

### DNS implementation

AdGuard

### Tunnel mode

Split-Tunnel

### Low-level settings

Bootstrap server:
Fallback server:
Blocking mode, etc:

### Issue Details

Steps to reproduce:
1. Have AdGuard Ablocker for IOS & AdGuard VPN for IOS enabled.
2. After so often the connection to private DNS cuts off and cuts of Network connection.
3. To get it working have to a least disable VPN and toggle wifi off the on again to refresh the connection.

### Expected Behavior

Should not cut off connection  to network 

### Actual Behavior

Network gets cut off.

### Additional Information

I will send the logs for this report.

**Comment by Batman2814** [comment]
Logs have been sent.

**Comment by Batman2814** [comment]
https://github.com/AdguardTeam/AdguardForiOS/issues/2189

Might be related to this also.

**Comment by maxikuzmin** [comment]
@Batman2814 can you give more details? What dns are you using? How often does the internet connection break? Does the connection break just happen after some time, are there any additional steps to break the internet connection?

**Comment by Batman2814** [comment]
I'm using AdGuard Private DNS. Protocol QUIC. All I know is it happens either when I come back from it. It also might have happened when using it, too. I always have AdGuard VPN for iOS enabled. It always seems to happen after some time.

**Comment by maxikuzmin** [comment]
@Batman2814 could you please clarify, with other DNS's this problem does not recur? Have you tried changing the protocol or DNS?

**Comment by Batman2814** [comment]
Good day. I have not yet attempted to modify the protocol or DNS settings. Could you please advise if I should proceed with either of these actions? If so, kindly specify whether I should prioritize changing the protocol to TLS or switching to AdGuard Public DNS within the AdGuard Adblocker Pro application for iOS. Your guidance would be greatly appreciated.

**Comment by maxikuzmin** [comment]
@Batman2814 we need to understand first whether the problem is with the individual DNS (or DNS protocol) server or with the DNS partition as a whole. So I ask you to try to select a different DNS with QUIC protocol, and also try with a different DNS (DOH or DOT protocol). Perhaps the problem could also be in Private DNS.

I tried also reproducing this problem but it did not happen again for me, it would be nice to get your debug logs as well

**Comment by Batman2814** [comment]
I would be happy to test that out and get back to you about the issue. I did send debug logs in the email. Could you please let me know what happened to that log?

**Comment by maxikuzmin** [comment]
@Batman2814 we got your logs, I meant logs with other DNS. By the way what about other DNS? Have you tried to check this problem with other DNS? After your update, after restarting the WIFI connection, did the problem recur again?

**Comment by Batman2814** [comment]
I have identified the potential source of the issue to be the AdGuard VPN iOS application. I conducted a test by disabling the AdGuard VPN, and the connection remained stable. However, upon activating the AdGuard VPN, the connection eventually terminated.

**Comment by AleksandrKalininAdGuard** [comment]
+ CRM #882294

**Comment by maxikuzmin** [comment]
@AleksandrKalininAdGuard unfortunately this is not a similar case

**Comment by maxikuzmin** [comment]
@Batman2814 did you manage to upload logs from both AdGuard + AdGuard VPN apps? 
Also please clarify how you configured Private DNS

**Comment by Batman2814** [comment]
I haven't yet yet, but I will do it.

**Comment by maxikuzmin** [comment]
@Batman2814 I tried to add my Private DNS (DOH), and at first I added it successfully, but then if I go into edit DNS and try to just re-save the Private DNS, it displays an error about incorrect upstream. Do you have the same behavior?

**Comment by Batman2814** [comment]
Could you please clarify what you mean by user rules for my DNS?

**Comment by maxikuzmin** [comment]
@Batman2814 on the AdGuard DNS website in the menu Server settings -> User Rules 
You may have rules affecting the network there

**Comment by Batman2814** [comment]
Logs have been Sent

**Comment by Batman2814** [comment]
Logs from Adguard VPN for iOS

**Comment by maxikuzmin** [comment]
@Batman2814 thanks, we will need some time to into it the logs

**Comment by maxikuzmin** [comment]
@Batman2814 we made some changes in version 4.5.3 
Could you please repeat the steps on version 4.5.3 and if the problem recurs send new logs?

**Comment by Batman2814** [comment]
Certainly, I will monitor the situation and inform you if the issue persists.

**Comment by maxikuzmin** [comment]
@Batman2814 any news?

**Comment by Batman2814** [comment]
Everything is good so far. AdGuard Private DNS hasn't cut off the connection with AdGuard VPN for iOS enabled.

**Comment by gpodder770** [comment]
I am also having this problem. Using any dns in native mode in the adguard app, causes internet to turn off when screen is turned off for like 1mins or more

but if I use the dns profile way to use dns then wifi is always on

using adguard dns but checked with nextdns also 

strange problem

**Comment by maxikuzmin** [comment]
@Batman2814 @gpodder770 is the problem still recurring in version 4.5.3?

**Comment by gpodder770** [comment]
> @Batman2814 @gpodder770 is the problem still recurring in version 4.5.3?

Yes for me

thus i am using apple configuration file

**Comment by maxikuzmin** [comment]
@gpodder770 so when you add any Custom DNS you have no internet connection when screen is turned off for like 1mins or more?

Could you please specify which addresses you are adding?

Also could you please send your logs?

Here's what we need you to do:

- Open Settings --> General --> Turn Advanced mode on --> Advanced settings --> Turn Debug logs switch on;
- Reproduce the issue and remember the exact time when it happened;
- Disable "DNS Privacy";
- Open the settings --> Support --> Export logs;
- System popup will be opened --> Choose "Mail" app
- Enter apple@adguard.com as a recipient;
- Mention this issue number in the subject and add the time the issue was reproduced at.

**Comment by maxikuzmin** [comment]
@gpodder770 any news?

**Comment by gpodder770** [comment]
> @gpodder770 any news?

I stopped using dns via adguard entirely 

using via profile system

**Comment by maxikuzmin** [comment]
@gpodder770 but we need your logs to figure out the problem. Could you please repeat the steps again as instructed and send us the logs?

**Comment by maxikuzmin** [comment]
@gpodder770 @Batman2814 for the moment, I’m closing this issue. You can reopen the issue in the future if the problem persists
